js/ui/grid_core/ui.grid.core.virtual_data_loader.js

```diff
@@ -1,411 +1,2 @@
-                        fireChanged(this, callBase, { changeType: 'append', items: [] });
-                        return;
-                    }
-                }
-            }
-
-            const cacheItem = { pageIndex: dataOptions.pageIndex(), itemsLength: dataOptions.items(true).length, itemsCount: this.itemsCount(true) };
-
-            if(this.option('scrolling.removeInvisiblePages') && isVirtualMode) {
-                removeInvisiblePages = this._cache.length > Math.max(getPreloadPageCount(this) + (this.option('scrolling.preloadEnabled') ? 1 : 0), 2);
-            } else {
-                processDelayChanged(this, callBase, { isDelayed: true });
-            }
-
-            let removeCacheItem;
-            if(beginPageIndex === dataOptions.pageIndex() + 1) {
-                if(removeInvisiblePages) {
-                    removeCacheItem = this._cache.pop();
-                }
-                changeType = 'prepend';
-                this._cache.unshift(cacheItem);
-            } else {
-                if(removeInvisiblePages) {
-                    removeCacheItem = this._cache.shift();
-                }
-                changeType = 'append';
-                this._cache.push(cacheItem);
-            }
-
-            const isDelayChanged = isVirtualMode && lastCacheLength === 0 && needTwoPagesLoading(this);
-            processChanged(this, callBase, this._cache.length > 1 ? changeType : undefined, isDelayChanged, removeCacheItem);
-            this._delayDeferred = this.load().done(() => {
-                if(processDelayChanged(this, callBase)) {
-                    this.load(); // needed for infinite scrolling when height is not defined
-                }
-            });
-        } else {
-            processChanged(this, callBase, e);
-        }
-    }
-
-    getDelayDeferred() {
-        return this._delayDeferred;
-    }
-
-    itemsCount(isBase) {
-        let itemsCount = 0;
-        const isVirtualMode = this._controller.isVirtualMode();
-
-        if(!isBase && isVirtualMode) {
-            this._cache.forEach((cacheItem) => {
-                itemsCount += cacheItem.itemsCount;
-            });
-        } else {
-            itemsCount = this._dataOptions.itemsCount();
-        }
-        return itemsCount;
-    }
-
-    virtualItemsCount() {
-        let pageIndex = getBeginPageIndex(this);
-        if(pageIndex < 0) {
-            pageIndex = this._dataOptions.pageIndex();
-        }
-        const beginItemsCount = pageIndex * this._dataOptions.pageSize();
-        const itemsCount = this._cache.length * this._dataOptions.pageSize();
-        const endItemsCount = Math.max(0, this._dataOptions.totalItemsCount() - itemsCount - beginItemsCount);
-        return {
-            begin: beginItemsCount,
-            end: endItemsCount
-        };
-    }
-
-    reset() {
-        this._loadingPageIndexes = {};
-        this._cache = [];
-    }
-}
+/** @deprecated */
```

Comment:
Let's change the js-doc comment to the usual comment
```javascript
/** @deprecated */
```
to
```javascript
// deprected
```

Because js-doc comments can cause some strange behaviour in our metadata-tools :(